### PR TITLE
FIX: reimplements chat audio into a service

### PR DESCRIPTION
**Pull request by jjaffeux**
This implementation attempts to be more resilient to background tab.

Notes:
- adds support for immediate arg in @debounce decorators
- fixes a bug in discourseDebounce which was not supporting immediate arg in tests
- chat-audio-manager has no tests as audio requires real user interaction and is hard to test reliably

### CODE REVIEW

File: plugins/chat/assets/javascripts/discourse/controllers/preferences-chat.js [hunk 1]
```diff
@@ -32,8 +35,7 @@ export default class PreferencesChatController extends Controller {
   @action
   onChangeChatSound(sound) {
     if (sound && !isTesting()) {
-      const audio = new Audio(CHAT_SOUNDS[sound]);
-      audio.play();
+      this.chatAudioManager.test(sound);
```
Comment (CvX) [inline plugins/chat/assets/javascripts/discourse/controllers/preferences-chat.js:38]: `test()`? In a `!isTesting()` block? That doesn't _sound_ right 😅

Comment (jjaffeux) [reply plugins/chat/assets/javascripts/discourse/controllers/preferences-chat.js:38]: Yes I had the same feeling, maybe `preview` ?

Comment (jjaffeux) [reply plugins/chat/assets/javascripts/discourse/controllers/preferences-chat.js:38]: `prehear`? :trollface:

Comment (jjaffeux) [reply plugins/chat/assets/javascripts/discourse/controllers/preferences-chat.js:38]: `sample` ?

Comment (jjaffeux) [reply plugins/chat/assets/javascripts/discourse/controllers/preferences-chat.js:38]: The alternative is some `play(sound, debounced = true)` which I'm not super fan of...

Comment (CvX) [reply plugins/chat/assets/javascripts/discourse/controllers/preferences-chat.js:38]: `playImmediately()`? does what it says 😅

Comment (jjaffeux) [reply plugins/chat/assets/javascripts/discourse/controllers/preferences-chat.js:38]: well... I guess that's an acceptable tradeoff :+1:

Comment (jjaffeux) [reply plugins/chat/assets/javascripts/discourse/controllers/preferences-chat.js]: done

**Review by CvX — APPROVED** [review]

**Pull request merged**